### PR TITLE
[net-http] Add RBS file for Net::HTTP

### PR DESCRIPTION
**Pull request by raosush**
# Description
- Added RBS file for Net::HTTP gem
- Added tests for Net::HTTP gem in test/stdlib/Net_test.rb
- Added a snippet in Rakefile to include URI module for validate task

Signed-off-by: Sushanth Sathesh Rao <rao.sush004@gmail.com>

### CODE REVIEW

**Comment by raosush** [comment]
- [ ] Add some more tests inside `Net_test.rb`

File: test/stdlib/Net_test.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+require_relative "test_helper"
+require "net/http"
+require "uri"
+
+class NetSingletonTest < Test::Unit::TestCase
+  include TypeAssertions
+
+  library "net-http"
```
Comment (soutaro) [inline test/stdlib/Net_test.rb:8]: Only one `#library` call is allowed, and it accepts multiple arguments! 💡 

```suggestion
  library "net-http", "uri"
```

Comment (raosush) [reply test/stdlib/Net_test.rb]: Thank you sir! 😃

**Comment by soutaro** [comment]
The cause of the CI failure is fixed on the master. Rebase on the master branch and `push -f`.

File: stdlib/net-http/0/net-http.rbs [hunk 2]
```diff
@@ -0,0 +1,1868 @@
...
+    #   }
+    #
+    #   # using iterator
+    #   http.request_get('/index.html') {|res|
+    #     res.read_body do |segment|
+    #       print segment
+    #     end
+    #   }
+    #
+    def read_body: (?untyped? dest) { (String parameter) -> untyped } -> String
```
Comment (soutaro) [inline stdlib/net-http/0/net-http.rbs:1444]: The doc says `<<` is used for writing the result to `dest`. So, we can write it in the type by interface.

```rbs
interface _Dest
  def <<: (String) -> void
end

......

def read_body: () -> String
             | (_Dest dest) -> String
             | () { (String) -> void } -> String
```

**Review by soutaro — CHANGES_REQUESTED** [review]

File: stdlib/net-http/0/net-http.rbs [hunk 3]
```diff
@@ -0,0 +1,1863 @@
...
+    # True if requests for this connection will be proxied
+    def proxy?: () -> bool
+
+    # True if the proxy for this connection is determined from the environment
+    def proxy_from_env?: () -> bool
+
+    def proxy_uri: () -> (nil | URI::Generic)
+
+    # The address of the proxy server, if one is configured.
+    def proxy_address: () -> (String | nil)
```
Comment (soutaro) [inline stdlib/net-http/0/net-http.rbs:743]: Merge to `attr_accessor proxy_address: String?`.

File: stdlib/net-http/0/net-http.rbs [hunk 4]
```diff
@@ -0,0 +1,1863 @@
...
+    #     response = http.get('/index.html')
+    #
+    #     # using block
+    #     File.open('result.txt', 'w') {|f|
+    #       http.get('/~foo/') do |str|
+    #         f.write str
+    #       end
+    #     }
+    #
+    def get: (String path, ?Hash[String, untyped]? initheader, ?bot? dest) ?{ (String body_segment) -> void } -> Net::HTTPResponse
```
Comment (soutaro) [inline stdlib/net-http/0/net-http.rbs:799]:
```suggestion
    def get: (String path, ?Hash[String, untyped] initheader, ?bot dest) ?{ (String body_segment) -> void } -> Net::HTTPResponse
```

1. Can `initheader` be `nil`?
2. Make `dest` non optional.

File: stdlib/net-http/0/net-http.rbs [hunk 5]
```diff
@@ -0,0 +1,1863 @@
...
+  end
+
+  module HTTPHeader
+    def initialize_http_header: (Hash[untyped, untyped] initheader) -> void
+
+    def size: () -> Integer
+
+    alias length size
+
+    type string_or_symbol = String | Symbol
```
Comment (soutaro) [inline stdlib/net-http/0/net-http.rbs:976]: I prefer using an abstract name, like `key`, but you can keep the name.

File: test/stdlib/Net_test.rb [hunk 6]
```diff
@@ -0,0 +1,512 @@
+require_relative "test_helper"
```
Comment (soutaro) [inline test/stdlib/Net_test.rb:1]: Rename the file to `Net_HTTP_test.rb` please.

File: stdlib/net-http/0/net-http.rbs [hunk 7]
```diff
@@ -0,0 +1,1863 @@
...
+    #   p response['Set-Cookie']
+    #     #=> "session=al98axx; expires=Fri, 31-Dec-1999 23:58:23, query=rubyscript; expires=Fri, 31-Dec-1999 23:58:23"
+    #
+    def get_fields: (string_or_symbol key) -> (nil | Array[String])
+
+    # Returns the header field corresponding to the case-insensitive key.
+    # Returns the default value +args+, or the result of the block, or
+    # raises an IndexError if there's no header field named +key+
+    # See Hash#fetch
+    def fetch: (string_or_symbol key, *untyped args) { (String) -> untyped } -> (String | Array[String])
```
Comment (soutaro) [inline stdlib/net-http/0/net-http.rbs:1026]:
```suggestion
    def fetch: (string_or_symbol key) -> String
             | (string_or_symbol key, untyped) -> untyped
             | (string_or_symbol key) { (String) -> untyped } -> untyped
```

I tested with some values:

```
irb(main):034:0> response.fetch("foo1") { [] }
=> ""
irb(main):035:0> response.fetch("foo1") { [] }
=> ""
irb(main):036:0> response.fetch("foo1", [])
=> ""
irb(main):037:0> response.fetch("foo1")
/Users/soutaro/.rbenv/versions/3.0.0/lib/ruby/3.0.0/net/http/header.rb:126:in `fetch': key not found: "foo1" (KeyError)
```

**Review by soutaro — CHANGES_REQUESTED** [review]
Reviewed the updated content. 🙏

**Review by soutaro — APPROVED** [review]
💪

**Pull request merged**